### PR TITLE
11.0 mig account analytic distribution

### PR DESCRIPTION
**Pull request by nadiaafa**

### CODE REVIEW

File: account_analytic_distribution/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_analytic_distribution/__init__.py]: With Python3 the coding utf-8 lines can be removed.

File: account_analytic_distribution/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_analytic_distribution/__manifest__.py]: Same here...

File: account_analytic_distribution/models/account_analytic_distribution.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
```
Comment (melroy89) [inline account_analytic_distribution/models/account_analytic_distribution.py]: Same..

**Review by melroy89 — CHANGES_REQUESTED** [review]
UTF-8 lines can be removed in v11 Python3. Just be sure to safe the files in UTF-8 encoding is enough.

File: account_analytic_distribution/__init__.py (hunk 1 above)
Comment (nadiaafa) [reply account_analytic_distribution/__init__.py]: @danger89 Thank you, i made the changes, could you check please.

**Review by melroy89 — APPROVED** [review]

**Comment by nadiaafa** [comment]
@danger89 , no other changes are required?

**Comment by yvaucher** [comment]
@nadiaafa Could you squash the commit for translation into a single one, please.

File: account_analytic_distribution/README.rst [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+Multiple analytic lines are created when the invoice or the entries are
+confirmed.
+
+You can continue using 'Analityc Account' field, but 'Analytic Distribution'
+has precedence if defined.
+
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/87/10.0
```
Comment (yvaucher) [inline account_analytic_distribution/README.rst]: 10.0 to change to 11.0

File: account_analytic_distribution/models/account_move_line.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        res = self._prepare_analytic_line()
+        for line_dict in res:
+            amount = (line_dict.get('amount') * rule.percent) / 100.0
+            line_dict['amount'] = amount
+            line_dict['account_id'] = rule.analytic_account_id.id
+        return res and res[0] or False
+
+    @api.multi
+    def create_analytic_lines(self):
+        super(AccountMoveLine, self).create_analytic_lines()
```
Comment (yvaucher) [inline account_analytic_distribution/models/account_move_line.py]: In python3 you can simplify it this way:
`super().create_analytic_lines()`

File: account_analytic_distribution/models/account_move_line.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        for line in self:
+            if line.analytic_distribution_id:
+                line.analytic_line_ids.unlink()
```
Comment (yvaucher) [inline account_analytic_distribution/models/account_move_line.py]: To make it a slightly more efficient, here we could write:

```python
ml_to_do = line.filtered('line.analytic_distribution_id')
al_to_delete = ml_to_do.mapped('analytic_line_ids')
# here we do a single DELETE in database for related analytic lines
al_to_delete.unlink()
for line in ml_to_do:
    for rule in line.analytic_distribution_id.rule_ids:
        values = line._analytic_line_distributed_prepare(rule)
        self.env['account.analytic.line'].create(values)
```

File: account_analytic_distribution/models/account_invoice.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    def inv_line_characteristic_hashcode(self, invoice_line):
+        code = super(AccountInvoice, self).inv_line_characteristic_hashcode(
```
Comment (yvaucher) [inline account_analytic_distribution/models/account_invoice.py]: super can be simplified with the form `super()`

https://docs.python.org/3.0/library/functions.html#super

**Review by yvaucher — APPROVED** [review]
Just some suggestions of improvement, otherwise LGTM

**Comment by nadiaafa** [comment]
@yvaucher thank you for the review, i made the changes.

File: account_analytic_distribution/models/account_move_line.py (hunk 6 above)
Comment (levkar) [reply account_analytic_distribution/models/account_move_line.py]: efficient but complicated

**Comment by yvaucher** [comment]
Thanks for the changes, still few things to fixup.


One line to fix to make travis happy with PEP8:

account_analytic_distribution/models/account_move_line.py:31:80: E501 line too long (83 > 79 characters)

Plus check travis one test is failing:

odoo.addons.account_analytic_distribution.tests.test_analytic_distribution

**Comment by pedrobaeza** [comment]
Superseeded by #184 as no answer

**Pull request closed without merging**